src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/TypeInfoResolverFunctionalTests.cs

```diff
@@ -0,0 +1,522 @@
+                            prop.Set = null;
+                        }
+                    }
+                }
+            });
+
+            JsonSerializerOptions options = new JsonSerializerOptions();
+            options.IncludeFields = true;
+            options.TypeInfoResolver = resolver;
+
+            TestClass originalObj = new TestClass()
+            {
+                TestField = "test value",
+                TestProperty = 42,
+            };
+
+            string json = JsonSerializer.Serialize(originalObj, options);
+            Assert.Equal(@"{""TestProperty"":42,""TestField"":""test value""}", json);
+
+            TestClass deserialized = JsonSerializer.Deserialize<TestClass>(json, options);
+            Assert.Equal(originalObj.TestField, deserialized.TestField);
+            Assert.Equal(0, deserialized.TestProperty);
+        }
+
+        [Fact]
+        public static void SetCustomNumberHandlingForAProperty()
+        {
+            DefaultJsonTypeInfoResolver resolver = new();
+            resolver.Modifiers.Add((ti) =>
+            {
+                if (ti.Type == typeof(TestClass))
+                {
+                    Assert.Equal(JsonTypeInfoKind.Object, ti.Kind);
+                    foreach (var prop in ti.Properties)
+                    {
+                        if (prop.Name == nameof(TestClass.TestProperty))
+                        {
+                            prop.NumberHandling = JsonNumberHandling.AllowReadingFromString | JsonNumberHandling.WriteAsString;
+                        }
+                    }
+                }
+            });
+
+            JsonSerializerOptions options = new JsonSerializerOptions();
+            options.IncludeFields = true;
+            options.TypeInfoResolver = resolver;
+
+            TestClass originalObj = new TestClass()
+            {
+                TestField = "test value",
+                TestProperty = 42,
+            };
+
+            string json = JsonSerializer.Serialize(originalObj, options);
+            Assert.Equal(@"{""TestProperty"":""42"",""TestField"":""test value""}", json);
+
+            TestClass deserialized = JsonSerializer.Deserialize<TestClass>(json, options);
+            Assert.Equal(originalObj.TestField, deserialized.TestField);
+            Assert.Equal(originalObj.TestProperty, deserialized.TestProperty);
+        }
+
+        [Fact]
+        public static void SetCustomConverterForAProperty()
+        {
+            DefaultJsonTypeInfoResolver resolver = new();
+            resolver.Modifiers.Add((ti) =>
+            {
+                if (ti.Type == typeof(TestClass))
+                {
+                    Assert.Equal(JsonTypeInfoKind.Object, ti.Kind);
+                    foreach (var prop in ti.Properties)
+                    {
+                        if (prop.Name == nameof(TestClass.TestProperty))
+                        {
+                            prop.CustomConverter = new PlusOneConverter();
```

Comment:
We should also have a test for setting a custom converter factory.

Reply:
there is unit test for that, I assume you mean functional test?